ffmpeg/__init__.py

```diff
@@ -74,29 +74,60 @@ class _FilterNode(_Node):
     def _get_filter(self):
         raise NotImplementedError()
 
+    def _get_params_from_dict(self, d):
+        params = ""
+        for k in self.kwargs:
+            params += k + "={}:".format(self.kwargs[k])
+        if len(params) > 0:
+            params = params[:-1]
+        return params
+
+    def _get_params_from_list(self, l):
+        return ":".join(["{}".format(i) for i in l])
+
+    def _get_filter_from_dict(self, d):
+        p = self._get_params_from_dict(d)
+        if len(p) > 0:
+            return self.NAME + "=" + p
+        return self.NAME
+
+
 
 class _TrimNode(_FilterNode):
     NAME = 'trim'
 
-    def __init__(self, parent, start_frame, end_frame, setpts='PTS-STARTPTS'):
+    def __init__(self, parent, **kwargs):
         super(_TrimNode, self).__init__([parent])
-        self.start_frame = start_frame
-        self.end_frame = end_frame
-        self.setpts = setpts
+        # if "setpts" not in kwargs:
+        #     kwargs["setpts"] = 'PTS-STARTPTS'
+        self.kwargs = kwargs
 
     def _get_filter(self):
-        return 'trim=start_frame={}:end_frame={},setpts={}'.format(self.start_frame, self.end_frame, self.setpts)
```

Comment:
I guess this can be simplified when `setpts` becomes its own filter; then we can just use`_get_filter_from_dict`.